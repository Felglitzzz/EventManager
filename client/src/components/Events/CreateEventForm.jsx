import React from 'react';
import { connect } from 'react-redux';
import toastr from 'toastr';
import PropTypes from 'prop-types';
import history from '../../helpers/history';
import { addNewEvent } from '../../actions/eventActions';
import { uploadToCloudinary } from '../../actions/imageActions';
import { loadCenters } from '../../actions/centerActions';
import EventForm from './Form/EventForm';
import Validate from '../../helpers/validations/Validate';
/**
 * class CreateEventForm
 */
class CreateEventForm extends React.Component {
  /**
   * constructor function
   * @param {object} props
   */
  constructor(props) {
    super(props);

    this.state = {
      eventData: {
        name: '',
        centerId: '',
        date: '',
        time: '',
        description: '',
        image: ''
      },
      chosenImage: '',
      errors: {},
      options: [],
      isLoading: false
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.redirectToEvents = this.redirectToEvents.bind(this);
    this.imageOnChange = this.imageOnChange.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.validate = this.validate.bind(this);
  }
  /**
   * Component lifecycle
   * @returns {func} componentWillMount
   */
  componentDidMount() {
    this.props.loadCenters();
  }
  /**
   * componentWillreceiveProps lifecycle
   * @param {object} nextProps
   * @returns {object} event
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.options) {
      this.setState({
        options: nextProps.options.loadedCenters.Centers
      });
    }
    if (nextProps.imageUrl) {
      this.setState({
        eventData: {
          ...this.state.eventData,
          image: nextProps.imageUrl.secure_url
        }
      });
    }
  }

  /**
   * @description handles on focus event
   * @method handleOnFocus
   *
   * @param { object } event - event object containing sign in details
   *
   * @returns { object } new sign in details state
   */
  handleFocus(event) {
    const field = event.target.name;
    const { errors } = this.state;
    errors[field] = '';
    this.setState({
      errors
    });
  }

  /**
   * onchange event function
   * @param {*} event
   * @returns {object} new state
   */
  onChange(event) {
    event.persist();
    const field = event.target.name;
    const { eventData } = this.state;
    eventData[field] = event.target.value;
    return this.setState({ eventData });
  }

  /**
   * @description handler for image upload - imageOnChange
   * @param {object} event
   * @returns {object} selected file
   */
  imageOnChange(event) {
    const chosenImage = event.target.files[0];
    const imageReader = new FileReader();
    if (chosenImage) {
      imageReader.onload = () => {
        const upload = new Image();
        upload.src = imageReader.result;
        upload.onload = () => {
          this.setState({
            uploadHeight: upload.height,
            uploadWidth: upload.width,
            chosenImage
          });
        };
      };
    }
    imageReader.readAsDataURL(chosenImage);
  }
  /**
   *@returns {void}
   */
  validate() {
    const { chosenImage, eventData } = this.state;
    const { errors, isValid } = Validate.createEvent(eventData, chosenImage);
    if (!isValid) {
      this.setState({ errors, isLoading: false });
      return errors;
    }
    this.setState({ errors: {}, isLoading: true });
  }
  /**
   * onSubmit event function
   * @param {*} event
   * @returns {void}
   */
  onSubmit(event) {
    event.persist();
    event.preventDefault();
    const errors = this.validate();
    if (errors) return;
    const { chosenImage } = this.state;
    this.props.uploadToCloudinary(chosenImage)
      .then(() => {
        const { eventData } = this.state;
        if (eventData.image) {
          this.props.addNewEvent(eventData)
            .then(() => this.redirectToEvents())
            .catch((error) => {
              this.setState({ isLoading: false });
              toastr.error(error);
            });
        }
      })
      .catch(() => {
        this.setState({ isLoading: false });
      });
  }

  /**
   * @returns {void}
   */
  redirectToEvents() {
    this.setState({ isLoading: false });
    toastr.success('Event Created');
    history.replace('/dashboard/events');
  }

  /**
  * @returns { react } component
  */
  render() {
    return (
      <div>
        <EventForm
          onSubmit={this.onSubmit}
          onChange={this.onChange}
          eventData={this.state.eventData}
          errors={this.state.errors}
          options={this.state.options}
          imageOnChange={this.imageOnChange}
          isLoading={this.state.isLoading}
          handleFocus={this.handleFocus}
        />
      </div>
    );
  }
}

CreateEventForm.propTypes = {
  addNewEvent: PropTypes.func.isRequired,
  loadCenters: PropTypes.func.isRequired,
  options: PropTypes.object.isRequired,
  uploadToCloudinary: PropTypes.func.isRequired,
  imageUrl: PropTypes.object,
};

/**
 * @param {object} state
 * @param {object} ownProps
 * @returns {object} state
 */
function mapStateToProps(state) {
  return {
    options: state.centers,
    imageUrl: state.images.image
  };
}
/**
 *
 * @param {func} dispatch
 * @returns {object} action
 */
function mapDispatchToProps(dispatch) {
  return {
    loadCenters: () => dispatch(loadCenters()),
    addNewEvent: eventData => dispatch(addNewEvent(eventData)),
    uploadToCloudinary: image => dispatch(uploadToCloudinary(image))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(CreateEventForm);
