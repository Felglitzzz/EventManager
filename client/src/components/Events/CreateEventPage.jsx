import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import history from '../../helpers/history';
import { addNewEvent } from '../../actions/eventActions';
import { uploadToCloudinary } from '../../actions/imageActions';
import { loadUnpaginatedCenters } from '../../actions/centerActions';
import EventForm from './Form/EventForm';
import Validate from '../../helpers/validations/Validate';
import Prompter from '../../helpers/Prompter';

/**
 * @description - Container class component for create event page
 *
 * @class CreateEventPage
 *
 * @extends {React.Component}
 */
export class CreateEventPage extends React.Component {
  /**
   * @description - creates an instance of CreateEventPage
   *
   * @constructor
   *
   * @param { props } props - contains create event component properties
   */
  constructor(props) {
    super(props);

    this.state = {
      eventData: {
        name: '',
        centerId: '',
        startDate: '',
        endDate: '',
        image: ''
      },
      chosenImage: '',
      chosenImageUrl: '',
      errors: {},
      options: [],
      isLoading: false
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.redirectToEvents = this.redirectToEvents.bind(this);
    this.imageOnChange = this.imageOnChange.bind(this);
    this.handleFocus = this.handleFocus.bind(this);
    this.validate = this.validate.bind(this);
  }

  /**
   * @description - Fetches centers after component mounts
   *
   * @memberof CreateEventPage
   *
   * @returns {void} Nothing
   */
  componentDidMount() {
    this.props.loadUnpaginatedCenters();
  }

  /**
   * @description - is invoked before the components receives new props
   *
   * @param {object} nextProps
   *
   * @returns {object} event
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.options.unPaginatedCenters) {
      this.setState({
        options: nextProps.options.unPaginatedCenters.centers
      });
    }
    if (nextProps.imageUrl) {
      this.setState({
        eventData: {
          ...this.state.eventData,
          image: nextProps.imageUrl.secure_url
        }
      });
    }
  }

  /**
   * @description handles on focus event
   *
   * @method handleOnFocus
   *
   * @param { object } event - event object containing create event form details
   *
   * @returns { void }
   */
  handleFocus(event) {
    const field = event.target.name;
    const { errors } = this.state;
    errors[field] = '';
    this.setState({
      errors
    });
  }

  /**
   * @description - handles create event form input change event
   *
   * @param {object} event
   *
   * @returns {void}
   */
  onChange(event) {
    event.persist();
    const field = event.target.name;
    const { eventData } = this.state;
    eventData[field] = event.target.value;
    return this.setState({ eventData });
  }

  /**
   * @description - handles image change event
   *
   * @param {object} event
   *
   * @returns {void}
   */
  imageOnChange(event) {
    const chosenImage = event.target.files[0];
    const imageReader = new FileReader();
    if (chosenImage) {
      imageReader.onload = () => {
        this.setState({
          chosenImage,
          chosenImageUrl: imageReader.result
        });
      };
    }
    imageReader.readAsDataURL(chosenImage);
  }

  /**
   * @description handles validation for create event input
   *
   * @method validate
   *
   * @param { object } event - event object
   *
   * @returns { object } error object and input validation status
   */
  validate() {
    const { chosenImage, eventData } = this.state;
    const { errors, isValid } = Validate.addEvent(eventData, chosenImage);
    if (!isValid) {
      this.setState({ errors, isLoading: false });
      return errors;
    }
    this.setState({ errors: {}, isLoading: true });
  }

  /**
   * @description - handles create-event form submission
   *
   * @param {object} event
   *
   * @returns {void}
   */
  onSubmit(event) {
    event.persist();
    event.preventDefault();
    const errors = this.validate();
    if (errors) return;
    const { chosenImage } = this.state;
    this.props.uploadToCloudinary(chosenImage)
      .then(() => {
        const { eventData } = this.state;
        if (eventData.image) {
          this.props.addNewEvent(eventData)
            .then(() => this.redirectToEvents())
            .catch((error) => {
              this.setState({ isLoading: false });
              Prompter.error(error);
            });
        }
      })
      .catch(() => {
        this.setState({ isLoading: false });
      });
  }

  /**
   * @description - handles redirect to all events page
   *
   * @returns {void}
   */
  redirectToEvents() {
    this.setState({ isLoading: false });
    Prompter.success('Event Created');
    history.replace('/dashboard/events');
  }

  /**
   * @description - renders create event form
   *
   * @returns {jsx} EventForm component
   */
  render() {
    return (
      <div>
        <EventForm
          errors={this.state.errors}
          eventData={this.state.eventData}
          handleFocus={this.handleFocus}
          imageOnChange={this.imageOnChange}
          isLoading={this.state.isLoading}
          onChange={this.onChange}
          onSubmit={this.onSubmit}
          options={this.state.options}
        />
      </div>
    );
  }
}

CreateEventPage.propTypes = {
  addNewEvent: PropTypes.func.isRequired,
  options: PropTypes.object.isRequired,
  uploadToCloudinary: PropTypes.func.isRequired,
  imageUrl: PropTypes.object,
  loadUnpaginatedCenters: PropTypes.func.isRequired
};

/**
 * @description maps redux state to props
 *
 * @param { object } state - holds redux state
 *
 * @return { object } props - returns mapped props from state
 */
const mapStateToProps = state => ({
  options: state.centerReducer,
  imageUrl: state.imageReducer.image
});

/**
 * @description maps action dispatched to props
 *
 * @param { object } dispatch - holds dispatchable actions
 *
 * @return { object } props - returns mapped props from dispatch action
 */
export const mapDispatchToProps = dispatch => ({
  loadUnpaginatedCenters: () => dispatch(loadUnpaginatedCenters()),
  addNewEvent: eventData => dispatch(addNewEvent(eventData)),
  uploadToCloudinary: image => dispatch(uploadToCloudinary(image))
});

export default connect(mapStateToProps, mapDispatchToProps)(CreateEventPage);
