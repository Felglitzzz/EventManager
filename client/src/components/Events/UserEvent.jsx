import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import { Link } from 'react-router-dom';
import history from '../../helpers/history';
/**
 * class UserEvent
 * @returns {object} react component
 */
const UserEvent = ({ events }) => (
  events.length === 0
    ?
    <h6>Loading events...</h6>
    :
    events.map((event, id) => (
      <section className="mt-3 mx-3" key={id}>
        <div className="container">
          <div className="row bg-white p-3">
            <div className="col-md-4">
              <div className="img-fluid">
                <img
                  className="img-fluid"
                  width="400"
                  height="264"
                  src={event.image}
                  alt={event.name}/>
              </div>
            </div>
            <div className="col-md-8">
              <div className="d-flex">
                <p className="lead text-secondary">{moment(event.date).format('MMMM Do YYYY')}</p>
                <p className="lead text-secondary ml-auto">{event.time}</p>
              </div>
              <h4 className="card-title list-group-item">{event.name}</h4>
              <p className="card-text list-group-item">{event.center.name}</p>
              <div className= "container pt-2 px-0">
                <ul className="pl-0">
                  <li className="d-flex justify-content-center">
                    <Link to={`/dashboard/events/${event.id}`} history={history}><p className="text-orange border-2 border-orange icon twitter mr-1">
                      <i className="fa fa-pencil"
                        aria-hidden="true" /></p></Link>
                    <a className="text-orange border-2 border-orange icon ig mr-1">
                      <i className="fa fa-trash"
                        aria-hidden="true" /></a>
                    <a
                      className="text-orange border-2 border-orange icon twitter mr-1" >
                      <i className="fa fa-ellipsis-h"
                        aria-hidden="true" /></a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </section>
    ))
);
export default UserEvent;

UserEvent.propTypes = {
  events: PropTypes.array.isRequired
};

