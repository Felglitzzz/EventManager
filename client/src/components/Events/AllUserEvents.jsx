import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import UserEvent from '../Events/UserEvent';
import { loadAllEvent } from '../../actions/eventActions';
import history from '../../helpers/history';
/**
 * class AllUserEvents
 * @return {array} component
 */
class AllUserEvents extends React.Component {
  /**
 * constructor function
 *
 * @param {object} props
 */
  constructor(props) {
    super(props);

    this.state = {
      event: []
    };

    this.redirectToEdit = this.redirectToEdit.bind(this);
  }
  /**
   * component lifecycle componentDidMount
   * @returns {void}
   */
  componentDidMount() {
    this.props.loadAllEvent();
  }

  /**
   * componentWillreceiveProps lifecycle
   *
   * @param {object} nextProps
   *
   * @returns {object} event
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.events) {
      this.setState({
        event: nextProps.events.event
      });
    }
  }
  /**
   * redirctToEdit function
   * @returns {void}
   */
  redirectToEdit() {
    history.push('/dashboard/events/:eventId');
  }
  /**
   * @returns { react } dashboard component
   */
  render() {
    return (
      < UserEvent
        events = {this.state.event}
        redirectToEdit = {this.redirectToEdit}
      />
    );
  }
}

AllUserEvents.propTypes = {
  events: PropTypes.object.isRequired,
  loadAllEvent: PropTypes.func.isRequired
};


/**
 * @param {object} state
 * @param {object} ownProps
 * @returns {object} loadedEvents
 */
const mapStateToProps = state => ({
  events: state.events.loadedEvents,
});

const mapDispatchToProps = dispatch => ({
  loadAllEvent: () => dispatch(loadAllEvent())
});

export default connect(mapStateToProps, mapDispatchToProps)(AllUserEvents);
