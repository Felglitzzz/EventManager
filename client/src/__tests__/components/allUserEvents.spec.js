import React from 'react';
import { shallow, mount } from 'enzyme';

import { AllUserEvents, mapDispatchToProps } from '../../components/Events/AllUserEvents';
import UserEvent from '../../components/Events/UserEvent';

const props = {
  loadAllEvent: () => Promise.resolve(),
  deleteEvent: () => Promise.resolve(),
  events: {
    events: {
      events: {
        count: 1,
        rows: [
          {
            id: 29,
            name: 'ooooooop',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1527113526/gtp7by5ybxchahusanxg.jpg',
            startDate: '2018-05-24T00:00:00.000Z',
            endDate: '2018-05-26T00:00:00.000Z',
            status: 'cancelled',
            userId: 1,
            centerId: 6,
            createdAt: '2018-05-23',
            updatedAt: '2018-06-02',
            center: {
              id: 6,
              name: 'Bespoke',
              location: 'Lekki'
            }
          }
        ]
      },
      meta: {
        pagination: {
          currentPageUrl: 'http://localhost:1991/api/v1/events?page=1',
          next: 'http://localhost:1991/api/v1/events?page=2',
          currentPage: 1,
          totalPages: 1,
          offset: 0,
          limit: 3
        }
      }
    }
  },
  history: {
    push: jest.fn(),
  },
  match: {
    params: {},
    path: '/dashboard/events'
  },
  location: {
    search: '?page=1'
  },
};

jest.mock('react-router-dom');


describe(' All User\'s Events Page', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<AllUserEvents {...props} />);
    expect(wrapper.find('section').exists).toBeTruthy();
  });

  it('should render stateless child component', () => {
    const childProps = {
      handleDelete: () => Promise.resolve(),
      events: {
        count: 1,
        rows: [
          {
            id: 29,
            name: 'ooooooop',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1527113526/gtp7by5ybxchahusanxg.jpg',
            startDate: '2018-05-24T00:00:00.000Z',
            endDate: '2018-05-26T00:00:00.000Z',
            status: 'cancelled',
            userId: 1,
            centerId: 6,
            createdAt: '2018-05-23',
            updatedAt: '2018-06-02',
            center: {
              id: 6,
              name: 'Bespoke',
              location: 'Lekki'
            }
          }
        ]
      },
    };
    const wrapper = shallow(<UserEvent {...childProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should test componentWillReceiveProps method', () => {
    const nextProps = {
      loadAllEvent: () => Promise.resolve(),
      deleteEvent: () => Promise.resolve(),
      events: {
        events: {
          events: {
            count: 1,
            rows: [
              {
                id: 29,
                name: 'ooooooop',
                image: 'https://res.cloudinary.com/felglitz/image/upload/v1527113526/gtp7by5ybxchahusanxg.jpg',
                startDate: '2018-05-24T00:00:00.000Z',
                endDate: '2018-05-26T00:00:00.000Z',
                status: 'cancelled',
                userId: 1,
                centerId: 6,
                createdAt: '2018-05-23',
                updatedAt: '2018-06-02',
                center: {
                  id: 6,
                  name: 'Bespoke',
                  location: 'Lekki'
                }
              }
            ]
          },
          meta: {
            pagination: {
              currentPageUrl: 'http://localhost:1991/api/v1/events?page=1',
              next: 'http://localhost:1991/api/v1/events?page=2',
              currentPage: 1,
              totalPages: 1,
              offset: 0,
              limit: 3
            }
          }
        }
      },
      history: {
        push: jest.fn(),
      },
      match: {
        params: {}
      },
      location: {
        search: '?page=1'
      }
    };
    const componentWillReceivePropsSpy = jest.spyOn(
      AllUserEvents.prototype,
      'componentWillReceiveProps'
    );
    const wrapper = shallow(<AllUserEvents {...props} />);
    wrapper.instance().componentWillReceiveProps(nextProps);
    expect(componentWillReceivePropsSpy).toHaveBeenCalledTimes(1);
  });

  it('should test the handleDelete method', () => {
    const wrapper = mount(<AllUserEvents {...props} />);
    wrapper.setState({
      eventsLoading: false,
      events: {
        count: 1,
        rows: [
          {
            id: 29,
            name: 'ooooooop',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1527113526/gtp7by5ybxchahusanxg.jpg',
            startDate: '2018-05-24T00:00:00.000Z',
            endDate: '2018-05-26T00:00:00.000Z',
            status: 'cancelled',
            userId: 1,
            centerId: 6,
            createdAt: '2018-05-23',
            updatedAt: '2018-06-02',
            center: {
              id: 6,
              name: 'Bespoke',
              location: 'Lekki'
            }
          }
        ]
      },
      pagination: {
        currentPageUrl: 'http://localhost:1991/api/v1/events?page=1',
        next: 'http://localhost:1991/api/v1/events?page=2',
        currentPage: 1,
        totalPages: 1,
        offset: 0,
        limit: 3
      }
    });
    expect(wrapper.find('section').exists).toBeTruthy();
    wrapper.instance().handleDelete({
      preventDefault: jest.fn(),
      target: {
        id: 6
      }
    });
    expect(wrapper.find('#deleteEventButton')).toHaveLength(1);
  });

  it('should test mapDispatchToProps method', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch).loadAllEvent).toBeTruthy();
    expect(mapDispatchToProps(dispatch).deleteEvent).toBeTruthy();
  });

  it('should test showNext method', () => {
    const history = { push: () => {} };

    const showNextSpy = jest.spyOn(AllUserEvents.prototype, 'showNext');
    const wrapper = mount(<AllUserEvents
      {...props}
      history = {history}
    />);
    wrapper.setState({
      pagination: {
        currentPageUrl: 'http://localhost:1991/api/v1/events?page=1',
        next: 'http://localhost:1991/api/v1/events?page=2',
        currentPage: 1,
        totalPages: 2,
        offset: 0,
        limit: 3
      },
      eventsLoading: false,
      events: {
        count: 1,
        rows: [
          {
            id: 29,
            name: 'ooooooop',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1527113526/gtp7by5ybxchahusanxg.jpg',
            startDate: '2018-05-24T00:00:00.000Z',
            endDate: '2018-05-26T00:00:00.000Z',
            status: 'cancelled',
            userId: 1,
            centerId: 6,
            createdAt: '2018-05-23',
            updatedAt: '2018-06-02',
            center: {
              id: 6,
              name: 'Bespoke',
              location: 'Lekki'
            }
          }
        ]
      },
      meta: {
        pagination: {
          currentPageUrl: 'http://localhost:1991/api/v1/events?page=1',
          next: 'http://localhost:1991/api/v1/events?page=2',
          currentPage: 1,
          totalPages: 1,
          offset: 0,
          limit: 3
        }
      }

    });
    wrapper.find('#showNext').simulate('click');
    expect(showNextSpy).toHaveBeenCalled();
  });

  it('should test showPrevious method', () => {
    const history = { push: () => {} };

    const showPreviousSpy = jest.spyOn(AllUserEvents.prototype, 'showPrevious');
    const wrapper = mount(<AllUserEvents
      {...props}
      history = {history}
    />);
    wrapper.setState({
      pagination: {
        previous: 'http://localhost:1991/api/v1/events?page=1',
        currentPageUrl: 'http://localhost:1991/api/v1/events?page=2',
        next: 'http://localhost:1991/api/v1/events?page=3',
        currentPage: 2,
        totalPages: 2,
        offset: 3,
        limit: 3
      },
      eventsLoading: false,
      events: {
        count: 1,
        rows: [
          {
            id: 29,
            name: 'ooooooop',
            image: 'https://res.cloudinary.com/felglitz/image/upload/v1527113526/gtp7by5ybxchahusanxg.jpg',
            startDate: '2018-05-24T00:00:00.000Z',
            endDate: '2018-05-26T00:00:00.000Z',
            status: 'cancelled',
            userId: 1,
            centerId: 6,
            createdAt: '2018-05-23',
            updatedAt: '2018-06-02',
            center: {
              id: 6,
              name: 'Bespoke',
              location: 'Lekki'
            }
          }
        ]
      },
      meta: {
        pagination: {
          currentPageUrl: 'http://localhost:1991/api/v1/events?page=1',
          next: 'http://localhost:1991/api/v1/events?page=2',
          currentPage: 1,
          totalPages: 1,
          offset: 0,
          limit: 3
        }
      }

    });
    wrapper.find('#showPrevious').simulate('click');
    expect(showPreviousSpy).toHaveBeenCalled();
  });

  it('should show no event page if there is no events', () => {
    const showNoEventsSpy = jest.spyOn(AllUserEvents.prototype, 'showNoEvents');
    const wrapper = shallow(<AllUserEvents {...props} />);

    wrapper.setState({
      eventsLoading: false,
      events: {}
    });

    const instance = wrapper.instance();
    instance.showNoEvents();
    expect(showNoEventsSpy).toHaveBeenCalled();
  });
});

