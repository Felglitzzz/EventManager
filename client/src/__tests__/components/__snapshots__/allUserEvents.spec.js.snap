// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` All User's Events Page should render stateless child component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserEvent
    events={
      Object {
        "count": 1,
        "rows": Array [
          Object {
            "center": Object {
              "id": 6,
              "location": "Lekki",
              "name": "Bespoke",
            },
            "centerId": 6,
            "createdAt": "2018-05-23",
            "endDate": "2018-05-26T00:00:00.000Z",
            "id": 29,
            "image": "https://res.cloudinary.com/felglitz/image/upload/v1527113526/gtp7by5ybxchahusanxg.jpg",
            "name": "ooooooop",
            "startDate": "2018-05-24T00:00:00.000Z",
            "status": "cancelled",
            "updatedAt": "2018-06-02",
            "userId": 1,
          },
        ],
      }
    }
    handleDelete={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <div
      className="col-sm-12 col-md-6 col-lg-4 my-2"
      id="userevent"
    >
      <div
        className="card-deck"
      >
        <div
          className="card"
        >
          <div
            className="img-fluid deck"
          >
            <img
              alt="ooooooop"
              className="img-fluid deck"
              height="264"
              src="https://res.cloudinary.com/felglitz/image/upload/v1527113526/gtp7by5ybxchahusanxg.jpg"
              width="400"
            />
          </div>
          <div
            className="card-body"
          >
            <div
              className="d-flex"
            >
              <p
                className="text-secondary"
              >
                May 24th 2018
              </p>
            </div>
            <hr
              className="my-1"
            />
            <h5
              className="card-title"
            >
              ooooooop
            </h5>
            <p
              className="card-text"
            >
              Bespoke
            </p>
          </div>
          <div
            className="stat text-white"
          >
            <span
              className="fa-stac pl-1"
            >
              <i
                className="fa fa-circle fa-stack text-danger"
              />
            </span>
            cancelled
          </div>
          <div
            className="overlay"
          >
            <div
              className="user-icons"
            >
              <div
                className="container pt-2 px-0"
              >
                <ul
                  className="pl-0"
                >
                  <li
                    className="d-flex justify-content-center"
                  >
                    <Link
                      id="editEventButton"
                      replace={false}
                      to="/dashboard/events/29"
                    >
                      <p
                        className="text-white bg-orange icon-lg mr-1 z-depth-2 z-depth-anim"
                      >
                        <i
                          aria-hidden="true"
                          className="fa fa-pencil fa-2x p-2"
                        />
                      </p>
                    </Link>
                    <a
                      id="deleteEventButton"
                    >
                      <p
                        className="text-white bg-orange icon-lg mr-1 z-depth-2 z-depth-anim"
                        id={29}
                        onClick={[Function]}
                      >
                        <i
                          aria-hidden="true"
                          className="fa fa-trash fa-2x p-2"
                          id={29}
                        />
                      </p>
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <div
        className="col-sm-12 col-md-6 col-lg-4 my-2"
        id="userevent"
      >
        <div
          className="card-deck"
        >
          <div
            className="card"
          >
            <div
              className="img-fluid deck"
            >
              <img
                alt="ooooooop"
                className="img-fluid deck"
                height="264"
                src="https://res.cloudinary.com/felglitz/image/upload/v1527113526/gtp7by5ybxchahusanxg.jpg"
                width="400"
              />
            </div>
            <div
              className="card-body"
            >
              <div
                className="d-flex"
              >
                <p
                  className="text-secondary"
                >
                  May 24th 2018
                </p>
              </div>
              <hr
                className="my-1"
              />
              <h5
                className="card-title"
              >
                ooooooop
              </h5>
              <p
                className="card-text"
              >
                Bespoke
              </p>
            </div>
            <div
              className="stat text-white"
            >
              <span
                className="fa-stac pl-1"
              >
                <i
                  className="fa fa-circle fa-stack text-danger"
                />
              </span>
              cancelled
            </div>
            <div
              className="overlay"
            >
              <div
                className="user-icons"
              >
                <div
                  className="container pt-2 px-0"
                >
                  <ul
                    className="pl-0"
                  >
                    <li
                      className="d-flex justify-content-center"
                    >
                      <Link
                        id="editEventButton"
                        replace={false}
                        to="/dashboard/events/29"
                      >
                        <p
                          className="text-white bg-orange icon-lg mr-1 z-depth-2 z-depth-anim"
                        >
                          <i
                            aria-hidden="true"
                            className="fa fa-pencil fa-2x p-2"
                          />
                        </p>
                      </Link>
                      <a
                        id="deleteEventButton"
                      >
                        <p
                          className="text-white bg-orange icon-lg mr-1 z-depth-2 z-depth-anim"
                          id={29}
                          onClick={[Function]}
                        >
                          <i
                            aria-hidden="true"
                            className="fa fa-trash fa-2x p-2"
                            id={29}
                          />
                        </p>
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
